spec/registry/unicode.md

```diff
@@ -0,0 +1,63 @@
+# MessageFormat 2.0 Unicode Registry
+
+The `u:` namespace is reserved for use by the Unicode Consortium.
```

Comment:
Perhaps quote the design document here?

```suggestion
This registry is for items in the namespace `u:` 
which is reserved for use by the Unicode Consortium. 
This registry can contain _functions_ or _options_. 
Implementations are not required to implement entries found in this registry. 

Items in this registry are stable and subject to stability guarantees. 
This registry might sometimes be used to incubate functionality before promotion to the RGI or default registry in a future release. 
In such cases, the `u:` namespace version is retained, but deprecated.
```

Reply:
The first paragraph is mostly fine, but as #634 is only a PR at this time, it's way too early to refer to the RGI, for instance, or talk about incubation. We can add this language later, when we get consensus on these processes and their names.

I would also find it a bit odd to say e.g. that this "can contain _functions_", when AFAIK no such `u:` function has been proposed by anyone.

Reply:
Permission to do something does not obligate us to actually do it. No one has proposed a `u:` function, but there's no reason to forbid them.

---

Are you disagreeing with the design in #634? More specifically, do you disagree with my thinking about how to organize the registries? I do think we should lock-step these efforts to some degree. We'll probably rename RGI somewhere along the way, for example. But I think we'll likely end up with the same concepts at least.

We could replace my line 9 above with something like:

> Items in this registry might be deprecated but will not be removed.

Reply:
I am in general agreement about having definitions for non-required functions and options, but I'm honestly not really sure what the current state of #634 is, and whether e.g. the registry development process added to the PR last Friday is final, or whether it's being iterated on. When we last discussed this on [a call in May](https://github.com/unicode-org/message-format-wg/blob/4514e880a8690ba2dd78a70696b4c89db93697ba/meetings/2024/notes-2024-05-06.md#pr-634), it was "Still in progress".

I'm also not sure whether the current design doc proposal's language "Implementations are not required to implement any values found in this registry and may adopt or ignore registry entries at their discretion." ought to be strengthened into something more like a SHOULD.

My preference would be to not lockstep processes that do not strictly need lockstep, and to focus first on the concrete (i.e. this PR), and use that experience to inform the registry maintenance plan. On which we ought to have a separate discussion, if/once it's in a stable state.

Reply:
Why functions?

The idea for the parameters is "u: parameters are potentially recognized / honored by all functions"
What would be the meaning of a "u:" function?

If we think "u:function" is a Unicode function, it is redundant.
All functions in the standard registry are Unicode functions, since this is a Unicode spec.